prototype/tests/test_app.py

```diff
@@ -0,0 +1,289 @@
+        """Test ordering single requirement."""
+        assert get_ordered_requirements(["download_speed"]) == ["download_speed"]
+
+
+class TestDefaultValues:
+    """Tests for default value generation."""
+
+    def test_download_default(self):
+        """Test default download speed."""
+        value = get_default_value_for_requirement("download_speed")
+        assert value > 0
+        assert isinstance(value, float)
+
+    def test_upload_default(self):
+        """Test default upload speed."""
+        value = get_default_value_for_requirement("upload_speed")
+        assert value > 0
+        assert isinstance(value, float)
+
+    def test_latency_default(self):
+        """Test default latency."""
+        value = get_default_value_for_requirement("latency")
+        assert value > 0
+        assert isinstance(value, float)
+
+    def test_packet_loss_default(self):
+        """Test default packet loss."""
+        value = get_default_value_for_requirement("packet_loss")
+        assert 0 <= value <= 1  # Should be decimal, not percentage
+        assert isinstance(value, float)
+
+    def test_unknown_default(self):
+        """Test default for unknown requirement."""
+        value = get_default_value_for_requirement("unknown_metric")
+        assert value == 0.0
+
+
+class TestInputConfiguration:
+    """Tests for input configuration generation."""
+
+    def test_download_config(self):
+        """Test download input configuration."""
+        config = get_requirement_input_config("download_speed")
+        assert "Mbps" in config["label"]
+        assert config["min_value"] >= 0
+        assert config["max_value"] > config["min_value"]
+        assert config["step"] > 0
+
+    def test_upload_config(self):
+        """Test upload input configuration."""
+        config = get_requirement_input_config("upload_speed")
+        assert "Mbps" in config["label"]
+        assert config["min_value"] >= 0
+        assert config["max_value"] > config["min_value"]
+
+    def test_latency_config(self):
+        """Test latency input configuration."""
+        config = get_requirement_input_config("latency")
+        assert "ms" in config["label"]
+        assert config["min_value"] >= 0
+        assert config["max_value"] > config["min_value"]
+
+    def test_packet_loss_config(self):
+        """Test packet loss input configuration."""
+        config = get_requirement_input_config("packet_loss")
+        assert "%" in config["label"]
+        assert config["format"] is not None
+        assert config["min_value"] == 0.0
+        assert config["max_value"] == 100.0
+
+
+class TestConfigFunctions:
+    """Tests for configuration-related functions."""
+
+    @pytest.fixture
+    def mock_config(self):
+        """Create a mock IQB configuration."""
+        return {
+            "use cases": {
+                "video_streaming": {
+                    "w": 1.0,
+                    "network requirements": {
```

Comment:
This is not a comment about the current pull request, rather a reminder for me for later. I think we should try to avoid spaces in JSON documents, because it's more idiomatic to have underscores and there are benefits in terms of JavaScript DX as well. I'll most likely circle around to this next time I notice/remember: I already cleaned up part of the JSONs but missed this part.